### PR TITLE
Fixed edge-case where step run is stored incompletely

### PR DESCRIPTION
**Pull request by AlexejPenner**
## Describe changes

I try to catch the error that we ran across today when a pipeline run fails and an incomplete step run is stored in the database which fails on retrieval. 

Whats missing for this PR is the ability to actually delete this step run/the pipeline.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/zen_stores/schemas/step_run_schemas.py [hunk 1]
```diff
@@ -212,11 +213,27 @@ def to_model(
...
-                json.loads(self.deployment.step_configurations)[self.name]
+            step_configuration = json.loads(
+                self.deployment.step_configurations
             )
-        elif self.step_configuration is not None:
+            if self.name in step_configuration:
+                full_step_config = Step.model_validate(
+                    step_configuration[self.name]
+                )
+            elif self.step_configuration:
```
Comment (schustmi) [inline src/zenml/zen_stores/schemas/step_run_schemas.py:226]: Isn't this case already handled below?

Maybe 
```python
  ...
elif not self.step_configuration:
  raise ValueError(...)
```

Comment (AlexejPenner) [reply src/zenml/zen_stores/schemas/step_run_schemas.py:226]: but it won't ever end up in the elif below once it goes into the else statement

Comment (schustmi) [reply src/zenml/zen_stores/schemas/step_run_schemas.py:226]: My suggestion is to get rid of the `else` end use an `elif` instead, see the code above

Comment (schustmi) [reply src/zenml/zen_stores/schemas/step_run_schemas.py:226]:
```python
if self.name in step_configuration:
    full_step_config = Step.model_validate(
        step_configuration[self.name]
    )
elif not self.step_configuration:
    raise ValueError(...)
```

**Review by schustmi — APPROVED** [review]

**Pull request merged**